htlcswitch/link.go

```diff
@@ -2404,6 +2438,64 @@ func (l *channelLink) handleUpstreamMsg(msg lnwire.Message) {
 		// Update the mailbox's feerate as well.
 		l.mailBox.SetFeeRate(fee)
 
+	case *lnwire.Stfu:
+		handleErr := func(err error) {
+			l.log.Error(err.Error())
+			l.fail(
+				LinkFailureError{
+					code:             ErrInvalidUpdate,
+					FailureAction:    LinkFailureDisconnect,
+					PermanentFailure: false,
+					Warning:          true,
+				},
+				err.Error(),
+			)
+		}
+
+		remoteOnLocal := l.channel.NumRemoteUpdatesPendingOnLocal()
+		remoteOnRemote := l.channel.NumRemoteUpdatesPendingOnRemote()
+		if remoteOnLocal > 0 || remoteOnRemote > 0 {
+			// This is invalid according to the spec, drop the
+			// connection.
+			err := fmt.Errorf(
+				"received stfu while remote updates pending "+
+					"on Local (%d) or Remote (%d)",
+				remoteOnLocal,
+				remoteOnRemote,
+			)
+			handleErr(err)
+
+			return
+		}
+
+		finished, err := l.quiescer.recvStfu()
+		if err != nil {
+			handleErr(err)
+			return
+		}
+
+		if finished {
+			return
+		}
+
```

Comment:
I would've thought we'd have some logic here to signal to the switch we're not eligible to forward. Maybe that's later on, still getting through the diff.

Reply:
This is handled implicitly in the EligibleToForward query.